### PR TITLE
Adding Override Yaml

### PR DESCRIPTION
**Pull request by Zelaskal**
Feature request from folks in lnet.  Essentially adds a base override that's processed after base.yaml but before any character-setup yamls.

### CODE REVIEW

**Comment by Dartellum** [comment]
I do not see the purpose of this. Any changes I want are handled in my specific yaml. Why would I want settings changed before my character-setup yaml is accessed?

**Comment by rpherbig** [comment]
I'm also curious what the use case is for this

**Comment by Zelaskal** [comment]
The chat in lnet was around having "globals" for all of your characters.  Some suggestions were made to modify the base.yaml, but I suggested this instead.  Apparently some people wanted consistency across characters for a handful of settings.

Somehow my mindstate check for outdoorsmanship made this PR too.  That'll teach me to branch before creating PRs.

**Comment by rpherbig** [comment]
I think the profile generator that @charleyc-dr made could be a more flexible way to solve this problem.

https://github.com/rpherbig/dr-scripts/blob/master/generate-profile.lic has a bunch of documentation in the header, and there are some examples in the `/profile-gen` directory.

What do you think?

**Comment by charleyc-dr** [comment]
Yep, the profile generator will definitely work for this.  Just make a `profile-gen/ALL-base-setup-include.yaml` file, move your character yamls into `profile-gen` as well, and add the line `%INCLUDE base-setup` to the top of each one.  Then running `;gen` will create yamls with your common info at the top.

File: dependency.lic [hunk 1]
```diff
@@ -552,6 +552,12 @@ class ScriptManager
     message_body += "\n"
     message_body += "Dependency Version: #{$DEPENDENCY_VERSION}"
     message_body += "\n"
+    
+    message_body += "\n****\n" * 3
+	overrideFile = File.path('./scripts/profiles/override.yaml')
```
Comment (rpherbig) [inline dependency.lic]: Will this file already be output by line 545?

File: outdoorsmanship.lic [hunk 2]
```diff
@@ -16,7 +16,9 @@ class Outdoorsmanship
   end
 
   def train_outdoorsmanship
-    if DRSkill.getrank('Outdoorsmanship') < 20
+    if DRSkill.getxp('Outdoorsmanship') >= 30
```
Comment (rpherbig) [inline outdoorsmanship.lic]: As noted in the PR, this change should removed

**Comment by rpherbig** [comment]
I can't really argue with the simplicity of this approach

File: dependency.lic [hunk 3]
```diff
@@ -552,6 +552,12 @@ class ScriptManager
...
+	message_body += overrideFile.to_s
+    message_body += "\n"
+    message_body += File.read(overrideFile)
```
Comment (jayRyan24) [inline dependency.lic]: Not sure about the override naming convention for this, it's not an end-all, I just find it possibly misleading. 

Possible alternatives:

1. common.yaml
2. base-common.yaml
3. common-settings.yaml

Just some examples.

**Comment by jayRyan24** [comment]
> It does make me wonder if there's a way to make ;gen more accessible/easier to use...

I wonder if you could put a YAML list inside of the generator, maybe some of the includes list the ../<name>-*.yamls and copies them into the gen folder. This could automate it and make it a tad easier.

**Comment by rcuhljr** [comment]
Seems fine, make sure to add the file contents to `;e help_me` output please. I do think it makes it a little more convoluted to find bugs with yaml because it adds another layer where settings can change but not my problems to debug anymore :)

**Comment by rcuhljr** [comment]
As far as naming, base-override.yaml or base-personal.yaml or something like that would be my preference.

**Comment by jayRyan24** [comment]
Yeah, it just needs something to say we're changing base, because all YAML files across the board are override files so to speak. 

If he doesn't respond to this and get it merged, I'll take it over, @rpherbig.

**Comment by rpherbig** [comment]
I'm on board with a different name to make it clearer that `base.yaml` is 'owned' by the repo (i.e. subject to change without notice) and `base-whateverThisIs.yaml` is 'owned' by the user (i.e. we will never overwrite it).

**Comment by Zelaskal** [comment]
Sorry folks, been a busy day.  Not sure if I will have time this weekend to put this back together or not. @jayRyan24 if you want to take this over you won't hurt my feelings.  I will close out this pull request either way.

**Pull request closed without merging**